src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -1126,6 +1126,174 @@ object Desugar {
 
           case n@PBreak(label) => unit(in.Break(label.map(x => x.name), nm.fetchBreakLabel(n, info))(src))
 
+          case n@PShortForRange(range, shorts, spec, body) =>
+            // is a block as it introduces new variables
+            unit(block(for {
+              exp <- goE(range.exp)
+              (elems, typ) = underlyingType(exp.typ) match {
+                case slice : in.SliceT => (slice.elems, slice)
+                case array : in.ArrayT => (array.elems, array)
+                case _ : in.MapT => violation("Maps are not supported yet in range")
+                case _ : in.StringT => violation("Strings are not supported yet in range")
+                case t => violation(s"Range not applicable to type $t")
+              }
+
+              // index is the place where we store the indices of the range expression
+              indexLeft <- leftOfAssignmentD(shorts(0))(in.IntT(Addressability.exclusiveVariable))
+              indexVar = in.Assignee.Var(indexLeft)
+              indexSrc = meta(shorts(0))
+              rangeSrc = meta(range)
+              indexAss = singleAss(indexVar, in.IntLit(0)(rangeSrc))(indexSrc)
+              incrIndex = singleAss(indexVar, in.Add(indexLeft, in.IntLit(1)(indexSrc))(indexSrc))(indexSrc)
+
+              rangeExpSrc = meta(range.exp)
+
+              // in go the range expression is only computed once before the iteration begins
```

Comment:
In Go, the expression inside range (let's call it `e`) is not necessarily evaluated if `len(e)` is constant (does not contain function calls and channel receives), which is probably meant as an optimization. I think that your solution is fine as long as in Gobra we guarantee that if len(src) is constant then src does not have side effects. Is this the case?

Furthermore, your solution is stricter then what Go allows. E.g., in Go, the following is totally fine:
```go
func main() {
	var x *[3][3]int
	fmt.Println(len(x[0]))
}
```
But your solution rejects it because it forces the evaluation of `x[0]` and `x` is `nil` here. Nonetheless, I am ok with this, as it looks to me that it is bad style to have these "ill-formed" expressions as parts of the program.